File: Aftas-FrontEnd/src/app/models/member.ts
import { Time } from "@angular/common";

export interface Member {
    id?: string;
    firstName?: string;
    lastName?: string;
    accessionDate?: Date;
    nationality?: string;
    identityNumber?: string;
    identityDocumentTypeEnum?: string;

}

export class CMember implements Member {
    id?: string;
    firstName?: string;
    lastName?: string;
    accessionDate?: Date;
    nationality?: string;
    identityNumber?: string;
    identityDocumentTypeEnum?: string;
}

File: AFTAS/src/main/java/com/example/aftas/domain/Member.java
package com.example.aftas.domain;

import com.example.aftas.domain.Enums.IdentityDocumentType;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.Date;
import java.util.List;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Member {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
//    @Column(name = "id", nullable = false)
    private Long id;

    private Integer number;

    private String firstName;

    private String lastName;

    private LocalDate accessionDate;

    private String nationality;

    private String identityNumber;

    private IdentityDocumentType identityDocumentTypeEnum;

    @JsonIgnore
    @OneToMany(mappedBy = "member")
    private List<Hunting> huntingList;

    @JsonIgnore
    @OneToMany(mappedBy = "member")
    private List<Ranking> rankingList;






}
